### PR TITLE
feat: add support for nullable response types

### LINKED ISSUE #60: Add complete support for returning nullable response types
In the existing solution when the response from the API Call is missing and the return type is marked as nullable then the response handler does not return the `null` value but instead fails by returning an empty string `''`.

Add the support for this above mentioned edge case, along with proper unit tests.

### PR DESCRIPTION
**Pull request by asadali214**
## What
This PR adds complete support for returning nullable response types using the `nullableType()` builder function to set the response types as nullable.

## Why
Previously response handler was throwing an exception or returning `''` when a response with no content is received.

Closes #60

## Type of change
Select multiple if applicable.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause a breaking change)
- [ ] Tests (adds or updates tests)
- [ ] Documentation (adds or updates documentation)
- [ ] Refactor (style improvements, performance improvements, code refactoring)
- [ ] Revert (reverts a commit)
- [ ] CI/Build (adds or updates a script, change in external dependencies)

## Dependency Change
N/A

## Breaking change
N/A

## Testing
Unit tests have been added to test this feature

## Checklist
- [X] My code follows the coding conventions
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added new unit tests

### CODE REVIEW

File: src/Response/Context.php [hunk 1]
```diff
@@ -65,6 +65,15 @@ public function isFailure(): bool
         return $statusCode !== min(max($statusCode, 200), 208); // [200,208] = HTTP OK
     }
 
+    /**
+     * Is response body missing.
+     */
+    public function isBodyMissing(): bool
+    {
+        $rawBody = $this->response->getRawBody();
+        return $rawBody === '';
```
Comment (sufyankhanrao) [inline src/Response/Context.php:74]: Blocker: As we agreed upon checking the whitespaces in the body as well therefore the rawBody should be trimmed here.

Comment (asadali214) [reply src/Response/Context.php]: Added changes

**Review by sufyankhanrao — APPROVED** [review]
